### PR TITLE
Sort apps in the app list case-insensitively

### LINKED ISSUE #846: App sorting should be case-insensitive
Poor draw.io is last on the list.

### PR DESCRIPTION
**Pull request by zarvox**
Fixes #846.

I took the liberty of cleaning up the methods I'd introduced in db.js to be
less Mongo-specific and to do matching and sorting in client code, rather than
using minimongo's limited capabilities for this (especially given the
complicated backwards-compatibility logic for appTitle and friends).

### CODE REVIEW

File: shell/packages/sandstorm-db/db.js [hunk 1]
```diff
@@ -690,13 +690,12 @@ _.extend(SandstormDb.prototype, {
   //   - It doesn't make it any easier for us to substitute a non-Mongo database in the future.
   //   - If the caller is just going to pass in a match-all query, we may be forcing Mongo into
   //     a slower path by using $and.
-  userGrains: function userGrains (user, query, aggregations) {
-    var filteredQuery = { $and: [ { userId: user}, query ] };
-    return this.collections.grains.find(filteredQuery, aggregations);
+  userGrains: function userGrains (user) {
```
Comment (kentonv) [inline shell/packages/sandstorm-db/db.js]: Is the comment above this line now obsolete?

File: shell/packages/sandstorm-db/db.js [hunk 2]
```diff
@@ -690,13 +690,12 @@ _.extend(SandstormDb.prototype, {
...
+    return this.collections.grains.find({ userId: user}, {});
```
Comment (kentonv) [inline shell/packages/sandstorm-db/db.js]: You can omit the empty second parameter (also on line 715).

File: shell/packages/sandstorm-db/db.js (hunk 1 above)
Comment (zarvox) [inline shell/packages/sandstorm-db/db.js]: Indeed.  Removed.

File: shell/packages/sandstorm-db/db.js (hunk 2 above)
Comment (zarvox) [inline shell/packages/sandstorm-db/db.js]: Done.

**Pull request merged**